<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>GWAS Tools &mdash; Validate 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="Validate 1.0 documentation" href="index.html" />
    <link rel="up" title="Getting Started with Validate" href="quickstart.html" />
    <link rel="next" title="Prediction Tools" href="prediction.html" />
    <link rel="prev" title="Uploading data to the Data Store and viewing in the Discovery Environment" href="datastore.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body role="document">
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="gwas-tools">
<h1>GWAS Tools<a class="headerlink" href="#gwas-tools" title="Permalink to this headline">¶</a></h1>
<div class="section" id="fastlmm">
<h2>FastLMM<a class="headerlink" href="#fastlmm" title="Permalink to this headline">¶</a></h2>
<p>The main genome wide association studies tool that we have used, FaST-LMM stands for Factored Spectrally Transformed Linear Mixed Models. It is a tool from Microsoft Research designed for analyses of very large data sets, and has been tested on data sets with over 120,000 individuals.
Running the Program</p>
<p>Please note that the original FaST-LMM executable destination is: /usr/bin/fastlmmc; thus, the executable can be launched from any directory on the instance. Simply type in fastlmmc into the terminal and a help menu will appear.</p>
<p>With that in mind, the program requires a minimum of three files to function: 1) A PEDMAP set of files, 2) a phenotype file corresponding to the PEDMAP set, and 3) a set of PLink formatted files to compute the genetic similarity matrix decomposition (does not need to be different from number 1).</p>
<p>The input flags you would use are as follows:</p>
<blockquote>
<div><ul class="simple">
<li>-file : Denotes the file name for the PLINK .ped/.map files</li>
<li>-bfile : Denotes the name for PLINK .bed/.bim/.fam files</li>
<li>-tfile : Denotes the name for PLINK .tped/.tfam files</li>
<li>-pheno : Denotes the name of the phenotype file (including extension)</li>
<li>-covar : Denotes the name of the covariate file (including file extension)</li>
<li>-out : The name of your final output file, which is placed into the same directory as your program and data unless otherwise specified</li>
</ul>
</div></blockquote>
<p>These are the bare minimum options needed to run FaST-LMM; however, some other options for considerations are:</p>
<blockquote>
<div><ul class="simple">
<li>-verboseOutput : use this flag to show more complex and detailed output; does not require a file to be named</li>
<li>-fileSim : The name of the PLINK set used for computing the genetic similarity matrix and its decomposition</li>
<li>-pValuePrintThreshold : Restricts the output file to only include SNPs with a p-value less than or equal to the specified threshold</li>
</ul>
</div></blockquote>
<p>An example of executing the FaST-LMM program might look like:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">fastlmmc</span> <span class="o">-</span><span class="n">verboseOutput</span> <span class="o">-</span><span class="n">bfile</span> <span class="n">toydata</span> <span class="o">-</span><span class="n">pheno</span> <span class="n">toydata</span><span class="o">.</span><span class="n">phe</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">covar</span> <span class="n">toydata</span><span class="o">.</span><span class="n">covar</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">out</span> <span class="n">MyResults</span><span class="o">.</span><span class="n">csv</span>
</pre></div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h3><a href="index.html">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="quickstart.html"> Quickstart Guide</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="accounts.html">Setting up your Accounts</a></li>
<li class="toctree-l2"><a class="reference internal" href="accounts.html#accessing-stampede">Accessing Stampede</a></li>
<li class="toctree-l2"><a class="reference internal" href="syngenta_tut.html">Tutorial for running Validate with sample data set</a></li>
<li class="toctree-l2"><a class="reference internal" href="alphasim.html">Generating a sample data set with Alphasim</a></li>
<li class="toctree-l2"><a class="reference internal" href="json.html">SLURM/JSON job submission guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="datastore.html">Using the Data Store</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">GWAS Tools</a></li>
<li class="toctree-l2"><a class="reference internal" href="prediction.html">Prediction</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="terms.html"> Useful Terms</a></li>
<li class="toctree-l1"><a class="reference internal" href="citations.html"> Citations &amp; Resources</a></li>
<li class="toctree-l1"><a class="reference internal" href="develop.html"> Further Development &amp; Getting help</a></li>
<li class="toctree-l1"><a class="reference internal" href="updating.html"> Updating the Docs</a></li>
</ul>

  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/gwas.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2016, UNCW Cyverse.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.4.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.8</a>
      
      |
      <a href="_sources/gwas.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>